src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfFeature.java

```diff
@@ -0,0 +1,1453 @@
+     * @return -1 if this < other; 0 if this == other; 1 if this > other
+     */
+    @Override
+    public int compareTo(GencodeGtfFeature other) {
+        Utils.nonNull(other);
+        return (baseData.featureOrderNumber - other.baseData.featureOrderNumber);
+    }
+
+    @Override
+    public boolean equals(Object that) {
+        if (that == null) {
+            return false;
+        }
+        else if ( this == that ) {
+            return true;
+        }
+
+        boolean isEqual = that instanceof GencodeGtfFeature;
+        if (isEqual) {
+            GencodeGtfFeature thatFeature = (GencodeGtfFeature) that;
+            Objects.equals(baseData, thatFeature.baseData);
+        }
+
+        return isEqual;
+    }
+
+    @Override
+    public int hashCode() {
+        return baseData != null ? baseData.hashCode() : 0;
+    }
+
+    /**
+     * Checks if {@code other} is contained within this {@link GencodeGtfFeature}.
+     * Comparison is made using {@link SimpleInterval#contains(Locatable)} ala {@link GencodeGtfFeatureBaseData#genomicPosition}
+     * @param other {@link GencodeGtfFeature} of which to check the bounds.
+     * @return true if {@code other} is contained within the bounds of this {@link GencodeGtfFeature}, false otherwise.
+     */
+    public boolean contains(GencodeGtfFeature other) {
+        Utils.nonNull(other);
+        return baseData.genomicPosition.contains(other.baseData.genomicPosition);
+    }
+
+    public void setFeatureOrderNumber(int featureOrderNumber) {
+        this.baseData.featureOrderNumber = featureOrderNumber;
+    }
+
+    // ================================================================================================
+
+    static public class OptionalField<T> {
+
+        private String name;
+        private T value;
+
+        public OptionalField(String name, T value) {
+            this.name = name;
+            this.value = value;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public void setValue(T value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+
+            StringBuilder sb = new StringBuilder();
+
+            sb.append(name);
+            sb.append(" ");
+
+            // We need to do some formatting for the numbers / non-numbers in the field:
+            String valueString = value.toString();
+            if ( NUMBER_PATTERN.matcher(valueString).matches() ) {
+                sb.append(valueString);
+                sb.append(";");
+            }
+            else {
+                sb.append("\"");
+                sb.append(valueString);
+                sb.append("\";");
+            }
+
+            return sb.toString();
+        }
+
+        @Override
+        public int hashCode() {
```

Comment:
Don't use `toString()` to generate the `hashCode()` -- `toString()` is just for display. Take the hashes of the individual fields and combine them, as you've done elsewhere.

Reply:
I guess I missed this one!

Fixed!